### PR TITLE
fix: show proper `req.url` in HTTPS mode

### PR DESCRIPTION
**Pull request by Skn0tt**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes https://github.com/netlify/cli/issues/5945.

Together with https://github.com/netlify/edge-functions-bootstrap/pull/294, this PR fixes what `req.url` shows in HTTPS mode.

It passes the original hostname and protocol into the Edge Function via `X-Forwarded-Host` and `X-Forwarded-Proto`. https://github.com/netlify/edge-functions-bootstrap/pull/294 reads this and shows it in `req.url`.

Bootstrap uses this URL for passthrough requests as well. In order for the Deno runtime to accept the self-signed certificate, we pass it into edge-bundler. This also allows us to remove the secondary HTTP server for passthrough requests.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/lib/edge-functions/bootstrap.mjs
Comment (Skn0tt) [inline src/lib/edge-functions/bootstrap.mjs:3]: We need to merge the ef-bootstrap PR first, so we can update the URL here.

File: src/utils/proxy.mjs
Comment (Skn0tt) [inline src/utils/proxy.mjs:747]: With this PR, we're passing the "potentially untrusted certificate" into Deno via the `--certs` flag.

This functionality was originally implemented in https://github.com/denoland/deno/pull/3972, and it works by adding the cert as a trusted root certificate to the HTTP client:

https://github.com/denoland/deno/pull/3972/files#diff-5c6fb275a66a1b02f99597f6e9b36297274abd4b16926b9fb2d412f94ee0700eR52

From my perspective, this should be fine. Maybe Eduardo can elaborate on the complications they had in mind when they're back.

**Review by sarahetter — APPROVED** [review]
LGTM, but it'd be great to wait for @eduardoboucas 's feedback on the comment above ^

File: src/utils/proxy.mjs
Comment (eduardoboucas) [reply src/utils/proxy.mjs:747]: Have you seen the PR that introduced that functionality? https://github.com/netlify/cli/pull/5409

**Review by eduardoboucas — CHANGES_REQUESTED** [review]
Since you're essentially reverting https://github.com/netlify/cli/pull/5409, can you add some colour to why you think we won't reintroduce the problems people were having with untrusted certificates? There's [a whole Slack thread](https://netlify.slack.com/archives/C0359548J07/p1674171123274979?thread_ts=1674141360.666799&cid=C0359548J07) (internal to Netlify only) with a discussion on why running Deno on HTTP would be preferable.

I understand that the behaviour described in https://github.com/netlify/cli/issues/5945 doesn't feel ideal, but is it breaking anything? I'd like to better understand what exactly we're looking to fix before we risk breaking other things.

**Comment by Skn0tt** [comment]
Wasn't aware of the Slack thread and the nuance around Deno requiring the CA cert. Agree that this isn't the right fix, in that case.

The thing that's broken is that `req.url` shows the wrong protocol and hostname. We can fix that using the `X-Forwarded-Proto` and `X-Forwarded-Host` headers, and that's probably the right way forward. Closing!

File: src/lib/edge-functions/proxy.mjs
Comment (eduardoboucas) [inline src/lib/edge-functions/proxy.mjs:173]: Can we make [the caller method](https://github.com/netlify/cli/blob/58b40b3af29a015434b99d854a841f146e291125/src/utils/proxy.mjs#L666) pass `settings.https` rather than using `req.socket.encrypted` here? That way we have a unified detection mechanism for whether we're running on HTTPS.

File: tests/integration/200.command.dev.test.cjs
Comment (eduardoboucas) [inline tests/integration/200.command.dev.test.cjs:245]: nit: I think using an error code here might come in handy.

```suggestion
                return new Response(error, { status: 500 })
```

File: src/lib/edge-functions/headers.mjs
Comment (eduardoboucas) [inline src/lib/edge-functions/headers.mjs:8]: Is this one still needed?

Comment (Skn0tt) [reply src/lib/edge-functions/headers.mjs:8]: We're still using `x-forwarded-proto` in the code. `x-forwarded-host` could be removed, if we wanted - but I think it's also fine to leave it in, so it's easier to discover the headers that ef-bootstrap takes in the future 🤷

File: src/lib/edge-functions/proxy.mjs
Comment (Skn0tt) [reply src/lib/edge-functions/proxy.mjs]: done in dfa73c9881fb44405d6fcf04e8ea738b0bf1c451

File: tests/integration/200.command.dev.test.cjs
Comment (Skn0tt) [reply tests/integration/200.command.dev.test.cjs]: done in 3c9a98c0dd51b778f1cea572b15bee107f2fc6df

File: src/lib/edge-functions/bootstrap.mjs
Comment (eduardoboucas) [inline src/lib/edge-functions/bootstrap.mjs:3]: Let's not forget to change this before merging.

**Review by eduardoboucas — CHANGES_REQUESTED** [review]
LGTM! I haven't approved because we're still using a deploy preview for the bootstrap URL. Once that's changed, we're good to roll.

File: src/lib/edge-functions/bootstrap.mjs
Comment (Skn0tt) [reply src/lib/edge-functions/bootstrap.mjs]: done in https://github.com/netlify/cli/pull/5968/commits/bb5f238fcd8ce2926679167ec1daf84844968490

**Review by eduardoboucas — APPROVED** [review]

**Pull request merged**
